File: Water Bottles.py
# https://leetcode.com/problems/water-bottles/?envType=daily-question&envId=2024-07-07
# Water Bottles

class Solution:
    
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        
        drink = 0
        empty = 0

        while numBottles > 0:
            
            drink += numBottles
            empty += numBottles

            numBottles = empty // numExchange
            empty %= numExchange

        return drink

File: Water Bottles.js
// https://leetcode.com/problems/water-bottles/?envType=daily-question&envId=2025-10-01
// Water Bottles

/**
 * @param {number} numBottles
 * @param {number} numExchange
 * @return {number}
 */
var numWaterBottles = function(numBottles, numExchange) {
    return numBottles + Math.floor((numBottles - 1) / (numExchange - 1));
};